### PR TITLE
Add corsDomains config property

### PR DESCRIPTION
**Pull request by nicolae-leonte-go**
We need to decide whether we want to apply the DomainFilter to every resource or not.

### CODE REVIEW

File: server/jersey-server/src/main/java/com/quorum/tessera/server/JerseyServer.java [hunk 1]
```diff
@@ -72,6 +72,8 @@ public void start() throws Exception {
         initParams.put("jersey.config.server.monitoring.statistics.enabled", "true");
         initParams.put("jersey.config.server.monitoring.enabled", "true");
         initParams.put("jersey.config.server.monitoring.statistics.mbeans.enabled", "true");
+        //custom properties
+        initParams.put("tessera.serverConfig", serverConfig);
```
Comment (melowe) [inline server/jersey-server/src/main/java/com/quorum/tessera/server/JerseyServer.java:76]: No.. We either access config in usual way . This second route to config and high jacking the initParams should be avoided

File: config/src/main/java/com/quorum/tessera/config/ServerConfig.java [hunk 2]
```diff
@@ -48,20 +48,25 @@
...
+
     public ServerConfig(final AppType app,
                         final boolean enabled,
                         final String serverAddress,
                         final CommunicationType communicationType,
                         final SslConfig sslConfig,
                         final InfluxConfig influxConfig,
-                        final String bindingAddress) {
+                        final String bindingAddress,
+                        final String corsDomains) {
```
Comment (melowe) [inline config/src/main/java/com/quorum/tessera/config/ServerConfig.java:61]: Domains is a list of things not just a string

Comment (melowe) [inline config/src/main/java/com/quorum/tessera/config/ServerConfig.java:61]: I'd also be tempted to not set this in constructor and default to null value. Will save having to change existing tests.

File: jaxrs-service/src/main/java/com/quorum/tessera/api/filter/DomainResponseFilter.java [hunk 3]
```diff
@@ -31,4 +43,16 @@ public void filter(ContainerRequestContext requestContext, ContainerResponseCont
 
     }
 
+
+    @Context
+    public void setConfiguration(Configuration configuration) {
+        this.corsDomains = initCorsDomains(configuration);
+    }
+
+    Set<String> initCorsDomains(Configuration configuration) {
```
Comment (melowe) [inline jaxrs-service/src/main/java/com/quorum/tessera/api/filter/DomainResponseFilter.java:52]: why is this protected?

File: config/src/main/java/com/quorum/tessera/config/ServerConfig.java (hunk 2 above)
Comment (nicolae-leonte-go) [reply config/src/main/java/com/quorum/tessera/config/ServerConfig.java:61]: yes, it is interpreted as a comma separated list of domains - see initCorsDomains.

Comment (melowe) [reply config/src/main/java/com/quorum/tessera/config/ServerConfig.java:61]: exactly.

**Pull request closed without merging**